src/pymodaq_plugins_thorlabs/daq_move_plugins/daq_move_KPZ101.py

```diff
@@ -0,0 +1,133 @@
+from pymodaq.utils.logger import set_logger, get_module_name
+
+logger = set_logger(get_module_name(__file__))
+
+
+class DAQ_Move_KPZ101(DAQ_Move_base):
+    """ Instrument plugin class for an actuator.
+
+    This object inherits all functionalities to communicate with PyMoDAQ’s DAQ_Move module through inheritance via
+    DAQ_Move_base. It makes a bridge between the DAQ_Move module and the Python wrapper of a particular instrument.
+
+    Attributes:
+    -----------
+    controller: object
+        The particular object that allow the communication with the hardware, in general a python wrapper around the
+         hardware library.
+
+    """
+    _controller_units = Piezo.default_units
+    is_multiaxes = True
+    _axes_names = {'1': 1}
+    _epsilon = 0.01
+    data_actuator_type = DataActuatorType.DataActuator
+    params = [
+                 {'title': 'Serial Number:', 'name': 'serial_number', 'type': 'list',
+                  'limits': serialnumbers_piezo, 'value': serialnumbers_piezo[0]}
+
+             ] + comon_parameters_fun(is_multiaxes, axes_names=_axes_names, epsilon=_epsilon)
+
+    def ini_attributes(self):
+        self.controller: Piezo = None
+        self._move_done = False
+
+    def get_actuator_value(self):
+        """Get the current value from the hardware with scaling conversion.
+
+        Returns
+        -------
+        float: The position obtained after scaling conversion.
+        """
+        pos = DataActuator(
+            data=self.controller.get_position(),
+            units=self.controller.get_units()
+        )
+        pos = self.get_position_with_scaling(pos)
+        return pos
+
+    def close(self):
+        """Terminate the communication protocol"""
+        if self.is_master:
+            self.controller.close()
+
+    def commit_settings(self, param: Parameter):
+        """Apply the consequences of a change of value in the detector settings
+
+        Parameters
+        ----------
+        param: Parameter
+            A given parameter (within detector_settings) whose value has been changed by the user
+        """
+        if param.name() == 'axis':
+            self.axis_unit = self.controller.get_units(self.axis_value)
+
+    def ini_stage(self, controller=None):
+        """Actuator communication initialization
+
+        Parameters
+        ----------
+        controller: (object)
+            custom object of a PyMoDAQ plugin (Slave case). None if only one actuator by controller (Master case)
+
+        Returns
+        -------
+        info: str
+        initialized: bool
+            False if initialization failed otherwise True
+        """
+
+        if self.is_master:
+            self.controller = Piezo()
+            self.controller.connect(self.settings['serial_number'])
+        else:
+            self.controller = controller
+
+        self.axis_unit = self.controller.get_units(self.axis_value)
```

Comment:
fine then see my comment above to use self.axis_unit (which is faster than talking to the controller)